wolfcrypt/src/hash.c

```diff
@@ -1219,6 +1219,11 @@ int wc_HashGetFlags(wc_HashAlg* hash, enum wc_HashType type, word32* flags)
             return MEMORY_E;
     #endif
 
+        /* Typically only important to hardware acceleration, but a good
+        ** practice: we'll make sure the md5 we start with does not have
+        ** any unexpected values in any of the properties. */
+        ForceZero(md5, sizeof(wc_Md5));
```

Comment:
Just use `XMEMSET` here please. The ForceZero is possible not as fast and is only required in places where there might be secrets that need cleared after use. This is just ensuring the data is zero'd, there shouldn't be any secrets left behind. Also this now duplicates work already done in some of the init functions. If using memset the compiler could optimize away the duplicate (not if ForceZero is used).

Reply:
This will have performance implications. My preference is to fix it in `wc_InitMd5`.

Reply:
I've reverted the changes to `hash.c` and moved the memory zeroize, now with `XMEMSET` instead of `ForceZero` into the respective `init`s.

Reply:
> duplicates work already done in some of the init functions

@dgarske regarding this topic,: Although the ESP32 SHA tests pass, the RSA fails when the apparently duplicate `XMEMSET` [here](https://github.com/gojimmypi/wolfssl/blob/6a17a140076333f7bff226f851133e6ba301a9f9/wolfcrypt/src/sha256.c#L778) which _should_ be ok to omit, as it is also [here](https://github.com/gojimmypi/wolfssl/blob/6a17a140076333f7bff226f851133e6ba301a9f9/wolfcrypt/src/sha256.c#L738) in the call to `InitSha256`.  Clearly there's more research to be done.